common/logger/src/libra_logger.rs

```diff
@@ -304,6 +304,38 @@ impl Writer for StderrWriter {
     }
 }
 
+/// A struct for writing logs to a file
+pub struct FileWriter {
+    log_file: std::sync::RwLock<std::fs::File>,
+}
+
+impl FileWriter {
+    pub fn new(log_file: std::path::PathBuf) -> Self {
+        let file = std::fs::OpenOptions::new()
+            .write(true)
+            .create(true)
+            .open(log_file)
+            .expect("Unable to open log file");
+        Self {
+            log_file: std::sync::RwLock::new(file),
+        }
+    }
+}
+
+impl Writer for FileWriter {
+    /// Write to file
+    fn write(&self, log: String) {
+        let result = self
+            .log_file
+            .write()
+            .unwrap()
+            .write_all(format!("{}\n", log).as_bytes());
```

Comment:
```suggestion
        let result = self
            .log_file
            .write()
            .unwrap()
            .write_all(log.as_bytes());
```

or

```suggestion
        let result = writeln!(self.log_file.write().unwrap(), "{}", log);
```

To avoid the allocation

Reply:
I figured I was missing something...

Reply:
Honestly the interface as it is right now should be tweaked to avoid an allocation but that's not your problem (its mine)